### PR TITLE
Use server-generated deviceId

### PR DESCRIPTION
**Pull request by richvdh**
This, finally, is what all of the refactoring over the last week has been in aid of. It makes sure that we actually use the same device_id for the crypto bits as we used to log in with. (We actually use the server-generated one to save us the hassle of generating our own everywhere).

Includes the changes in https://github.com/matrix-org/matrix-react-sdk/pull/408

### CODE REVIEW

File: src/Lifecycle.js [hunk 1]
```diff
@@ -128,6 +128,7 @@ function _loginWithToken(queryParams) {
         console.log("Logged in with token");
         setLoggedIn({
             userId: data.user_id,
+            deviceId: data.device_id,
```
Comment (dbkr) [inline src/Lifecycle.js]: Presumably `deviceId` will end up being `undefined` if you're using a server that doesn't support giving device IDs? Presumably only fairly recent synapses do? Should we disable crypto in this case?

Comment (richvdh) [inline src/Lifecycle.js]: yes, and yes we do. from the MatrixClient constructor:

```
    if (CRYPTO_ENABLED && opts.sessionStore !== null &&
            userId !== null && this.deviceId !== null) {
        // do crypto stuff
    }
```

Comment (dbkr) [inline src/Lifecycle.js]: Does this definitely work - I have a feeling going through localstorage will cause your `undefined` to turn into  the string `"undefined"`.

**Comment by richvdh** [comment]
You make a good point. PTAL.

**Pull request merged**